Improve aviation calculators with consistent styling and functionality

andures opened the pull request:
## Changes Made

This PR improves the aviation calculators with consistent design and functionality:

### Functionality Improvements:
- Implemented enhanced unit conversion between meters and feet
- Standardized Load/Save Parameters functionality
- Standardized JSON format for saved parameters
- Improved error handling and data validation
- Added functionality to copy results as HTML table
- Fully implemented the VSS/OCS calculator functionality

### Visual Improvements:
- Consistent dark mode support across all pages
- Coherent visual styles for buttons, inputs, and results
- Improved responsive design
- Enhanced results visualization

### Modified Files:
- ISA_deviation.html
- TAS.html
- rate_turn.html
- Profile_Check.html
- dark-mode.css (new file)

### Screenshots:
![image](https://github.com/user-attachments/assets/9cd61c3d-cc0b-4b94-9a73-7a06d36e89e4)
![image](https://github.com/user-attachments/assets/b81183d3-1a03-42ca-a9fb-c01cf7ad974c)
![image](https://github.com/user-attachments/assets/6ba510a8-aa88-45d6-ad02-a7ae91f56d84)
![image](https://github.com/user-attachments/assets/daabf5ae-3fa3-4d67-829b-05070d8a9d37)

flyght7-admin commented:
@andures Can we keep the input box locations as in the original e.g.
![image](https://github.com/user-attachments/assets/bbfe5bd0-8ce0-4a7d-af4c-b4fbe52f9ec2)

Save/Load buttons and Calculate look nice. Also let's reduce the v to 0.1 as its a beta release for the moment

andures commented:
## Changes Made

-  keep the input box locations as in the original
- reduce the v to 0.1 as its a beta
![Macbook-Air-1559x975](https://github.com/user-attachments/assets/63d91ffd-1c16-4cac-9453-ac24b028357a)

### 1. Enhanced Touch Targets

- Increased minimum height of buttons and sidebar items to 48px
- Added additional padding to interactive elements
- Increased spacing between sidebar items to prevent accidental taps
- Optimized the dark mode toggle for touch interaction


### 2. Swipe Gesture Support

- Added left-edge swipe detection to open the sidebar
- Implemented swipe-to-close functionality for the sidebar
- Added content area swipe detection for opening/closing the sidebar
- Created a dedicated swipe area on the left edge for easier sidebar access


```javascript
// New function to initialize touch events
function initTouchEvents() {
    const swipeArea = document.getElementById('swipeArea');
    const sidebar = document.getElementById('sidebar');
    let touchStartX = 0;
    let touchEndX = 0;
    
    // Edge swipe to open sidebar
    swipeArea.addEventListener('touchstart', function(e) {
        touchStartX = e.changedTouches[0].screenX;
    }, { passive: true });
    
    swipeArea.addEventListener('touchend', function(e) {
        touchEndX = e.changedTouches[0].screenX;
        handleSwipeGesture(touchStartX, touchEndX);
    }, { passive: true });
    
    // Additional touch event handlers for sidebar and content area
    // ...
}
```

### 3. Haptic Feedback

- Added vibration feedback for UI interactions using the Vibration API
- Implemented subtle haptic feedback when:

- Opening/closing the sidebar
- Selecting a calculator page
- Toggling dark mode





### 4. Visual Feedback Improvements

- Added scale transform effect on button press for visual feedback
- Removed default tap highlight color for cleaner UI
- Prevented text selection on touch for better user experience


```css
/* CSS for improved touch feedback */
@media (hover: none) and (pointer: coarse) {
    /* Prevent accidental text selection on touch */
    * {
        -webkit-tap-highlight-color: transparent;
        -webkit-touch-callout: none;
    }
    
    /* Add visible active state for touch feedback */
    button:active, .sidebar-item:active {
        transform: scale(0.98);
        transition: transform 0.1s;
    }
}
```

### 5. Scroll Handling

- Improved iframe content scrolling to prevent interference with main page
- Added event listeners to properly handle touch events between iframe and parent


### 6. New UI Elements

- Added a dedicated swipe area along the left edge for easier sidebar access
- Optimized the mobile header for touch interaction


## Technical Implementation

- Used CSS media queries targeting touch devices: `@media (hover: none) and (pointer: coarse)`
- Implemented touch event listeners with passive flag for better performance
- Added transform transitions for smooth visual feedback
- Used the Vibration API with fallback for devices without support


## Browser Compatibility

- Tested on iOS Safari, Chrome for Android, and iPadOS
- Compatible with all modern browsers that support touch events
- Graceful fallback for devices without vibration support
![image](https://github.com/user-attachments/assets/c2db04a3-2f7f-4fc1-86fc-caa7a1efaeb4)


## Performance Considerations

- Used passive event listeners to prevent scroll jank
- Optimized animations for smooth performance on lower-end devices
- Minimized DOM operations during touch handling


This update significantly improves the user experience on tablets and other touch-enabled devices while maintaining full functionality for desktop users.
